src/App.tsx

```diff
@@ -1,61 +1,100 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { TodoList } from './components/TodoList';
+
+import { todosWithUsers } from './utils/getTodoesWithUsers';
+import { getRandomId } from './utils/getRandomId';
+import { findUserById } from './utils/findUserById';
 
 export const App = () => {
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setUserIdError] = useState(false);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [todos, setTodos] = useState(todosWithUsers);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodo = {
+      userId,
+      id: getRandomId({ todos }),
+      title,
+      completed: false,
+      user: findUserById(userId),
+    };
+
+    setTodos([...todos, newTodo]);
+    setTitle('');
+    setUserId(0);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            value={title}
+            onChange={handleTitleChange}
+            placeholder="Enter todo title"
+          />
+          {hasTitleError && <span className="error">Please enter a title</span>}
```

Comment:
error messages can also be output in variables
```
const EMPTY_TITLE_ERROR = 'Please enter a title';
const EMPTY_USER_ERROR = 'Please choose a user';
```

```suggestion
          {hasTitleError && <span className="error">{EMPTY_TITLE_ERROR}</span>}
```